core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -1522,6 +1576,10 @@ public void trimUnreferencedReaders() throws IOException {
                 // we're shutdown potentially on some tragic event, don't delete anything
                 return;
             }
+            if (readers.isEmpty()) {
```

Comment:
methods should have a single return statement IMO, everything else is more complex in 99% of the cases.